generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java

```diff
@@ -25,47 +25,70 @@ a copy of this software and associated documentation files (the
 
 package com.pholser.junit.quickcheck.generator.java.util;
 
+import com.pholser.junit.quickcheck.generator.GenerationStatus;
+import com.pholser.junit.quickcheck.generator.Generator;
+import com.pholser.junit.quickcheck.generator.java.lang.AbstractStringGenerator;
+import com.pholser.junit.quickcheck.generator.java.lang.Encoded;
+import com.pholser.junit.quickcheck.generator.java.lang.StringGenerator;
+import com.pholser.junit.quickcheck.random.SourceOfRandomness;
+
 import java.util.Dictionary;
 import java.util.HashSet;
 import java.util.Hashtable;
 import java.util.Map;
 import java.util.Properties;
 import java.util.Set;
 
-import com.pholser.junit.quickcheck.generator.GenerationStatus;
-import com.pholser.junit.quickcheck.generator.Generator;
-import com.pholser.junit.quickcheck.generator.java.lang.StringGenerator;
-import com.pholser.junit.quickcheck.random.SourceOfRandomness;
-
-import static java.util.Arrays.*;
+import static java.util.Arrays.asList;
 
 /**
  * Produces values of type {@link Properties}.
+ *
  */
 public class PropertiesGenerator extends Generator<Properties> {
-    private final StringGenerator stringGenerator = new StringGenerator();
+    private AbstractStringGenerator stringGenerator;
 
-    public PropertiesGenerator() {
+    private PropertiesGenerator(AbstractStringGenerator stringGenerator) {
         super(Properties.class);
+        this.stringGenerator = stringGenerator;
+    }
+
+    /**
+     * By default, the internal {@link AbstractStringGenerator} instance is a {@link StringGenerator}.
+     *
+     */
+    public PropertiesGenerator() {
+        this(new StringGenerator());
+    }
+
+    /**
+     * If string generation for property keys and values should limited to a given {@link Encoded.InCharset},
+     * then use this constructor.
+     */
+    public PropertiesGenerator(Encoded.InCharset charset) {
+        this(new Encoded());
+        ((Encoded) stringGenerator).configure(charset);
     }
 
-    @Override public Properties generate(SourceOfRandomness random, GenerationStatus status) {
+    @Override
+    public Properties generate(SourceOfRandomness random, GenerationStatus status) {
         int size = status.size();
 
         Properties properties = new Properties();
         for (int i = 0; i < size; ++i) {
             properties.setProperty(
-                stringGenerator.generate(random, status),
-                stringGenerator.generate(random, status));
+                    stringGenerator.generate(random, status),
```

Comment:
@brentvukmer Yeah, I'd prefer keeping both scope indent and long-line indent at four spaces. Thanks!